crates/ruff/src/rules/pylint/rules/repeated_equality_comparison_target.rs

```diff
@@ -65,27 +69,45 @@ pub(crate) fn repeated_equality_comparison_target(checker: &mut Checker, bool_op
         return;
     }
 
-    let mut left_to_comparators: FxHashMap<HashableExpr, (usize, Vec<&Expr>)> =
-        FxHashMap::with_capacity_and_hasher(bool_op.values.len(), BuildHasherDefault::default());
+    let mut value_to_comparators: FxHashMap<HashableExpr, (usize, Vec<&Expr>)> =
+        FxHashMap::with_capacity_and_hasher(
+            bool_op.values.len() * 2,
+            BuildHasherDefault::default(),
+        );
+
     for value in &bool_op.values {
-        if let Expr::Compare(ExprCompare {
+        // Enforced via `is_allowed_value`.
+        let Expr::Compare(ast::ExprCompare {
             left, comparators, ..
         }) = value
-        {
-            let (count, matches) = left_to_comparators
-                .entry(left.deref().into())
-                .or_insert_with(|| (0, Vec::new()));
-            *count += 1;
-            matches.extend(comparators);
-        }
+        else {
+            return;
+        };
+
+        // Enforced via `is_allowed_value`.
+        let [right] = comparators.as_slice() else {
+            return;
+        };
+
+        let (left_count, left_matches) = value_to_comparators
+            .entry(left.deref().into())
+            .or_insert_with(|| (0, Vec::new()));
+        *left_count += 1;
+        left_matches.push(right);
+
+        let (right_count, right_matches) = value_to_comparators
+            .entry(right.into())
+            .or_insert_with(|| (0, Vec::new()));
+        *right_count += 1;
+        right_matches.push(left);
     }
 
-    for (left, (count, comparators)) in left_to_comparators {
+    for (value, (count, comparators)) in value_to_comparators {
```

Comment:
@tjkuson - What do you think of this? It's a bit closer to the previous version: we just track the comparators with the thing they're compared against (and this time, including the RHS), then raise if any comparator is used multiple times.